src/main/kotlin/ecommerce/repository/CartRepository.kt

```diff
@@ -0,0 +1,149 @@
+            )
+        }
+
+    fun save(cart: Cart): Cart {
+        val sql =
+            """
+            INSERT INTO cart (member_id, product_id, quantity, added_at)
+            VALUES (?, ?, ?, ?)
+            """.trimIndent()
+
+        val addedAt = cart.addedAt ?: LocalDateTime.now()
+
+        jdbc.update(
+            sql,
+            cart.memberId,
+            cart.productId,
+            cart.quantity,
+            Timestamp.valueOf(addedAt),
+        )
+        return cart.copy(addedAt = addedAt)
+    }
+
+    fun findByUserId(userId: Long): List<Cart> {
+        val sql = "SELECT * FROM cart WHERE member_id = ?"
+        return jdbc.query(sql, cartRowMapper, userId)
+    }
+
+    fun findByUserIdAndProductId(
+        userId: Long,
+        productId: Long,
+    ): Cart? {
+        val sql = "SELECT * FROM cart WHERE member_id = ? AND product_id = ?"
+        val results = jdbc.query(sql, cartRowMapper, userId, productId)
+        return results.firstOrNull()
+    }
+
+    fun update(cart: Cart): Cart {
+        val sql =
+            """
+            UPDATE cart 
+            SET quantity = ?, added_at = ?
+            WHERE member_id = ? AND product_id = ?
+            """.trimIndent()
+
+        val addedAt = cart.addedAt ?: LocalDateTime.now()
+
+        jdbc.update(
+            sql,
+            cart.quantity,
+            Timestamp.valueOf(addedAt),
+            cart.memberId,
+            cart.productId,
+        )
+        return cart.copy(addedAt = addedAt)
+    }
+
+    fun deleteByUserIdAndProductId(
+        userId: Long,
+        productId: Long,
+    ) {
+        val sql = "DELETE FROM cart WHERE member_id = ? AND product_id = ?"
+        jdbc.update(sql, userId, productId)
+    }
+
+    fun deleteByUserId(userId: Long) {
+        val sql = "DELETE FROM cart WHERE member_id = ?"
+        jdbc.update(sql, userId)
+    }
+
+    fun findAll(): List<Cart> {
+        val sql = "SELECT * FROM cart"
+        return jdbc.query(sql, cartRowMapper)
+    }
+
+    fun findTop5MostAddedProductsLast30Days(): List<Map<String, Any?>> {
+        val sql =
+            """
+            SELECT 
+                p.name as productName,
+                SUM(c.quantity) as addedCount,
+                MAX(c.added_at) as mostRecentAdded
+            FROM cart c
+            JOIN products p ON c.product_id = p.id
+            WHERE c.added_at >= DATEADD('DAY', -30, CURRENT_TIMESTAMP)
+            GROUP BY c.product_id, p.name
+            ORDER BY SUM(c.quantity) DESC, MAX(c.added_at) DESC
+            LIMIT 5
+            """.trimIndent()
+
+        return try {
+            jdbc.query(sql) { rs, _ ->
+                mapOf(
+                    "productName" to rs.getString("productName"),
+                    "addedCount" to rs.getInt("addedCount"),
+                    "mostRecentAdded" to rs.getTimestamp("mostRecentAdded")?.toLocalDateTime(),
+                )
```

Comment:
As mentioned elsewhere, it would be better to create a response DTO for this part.

Reply:
@codebrew42  Could you please apply this here? 🙏